server/routerlicious/packages/services-client/src/auth.ts

```diff
@@ -33,7 +33,7 @@ export function generateToken(
         user,
     };
 
-    return jwt.sign(claims, key);
+    return jsrsasign.jws.JWS.sign(null, JSON.stringify({ alg:"HS256", typ: "JWT" }), claims, key);
```

Comment:
Services also takes a dependency on this generateToken() function. I am assuming using jsrassign in Node.js is okay?

Reply:
It should, how can we test it?

Reply:
I did a quick runkit test (https://npm.runkit.com/jsrsasign) and seems to be working with a warning "(node:1) [DEP0005] DeprecationWarning: Buffer() is depreca…, Buffer.allocUnsafe(), or Buffer.from() methods instead".

I was curious about the security aspect when running on service. I personally prefer using well known vetted libraries when it comes to security. But if the jwt.io people recommend this when running on node, I am okay with this too.

Reply:
We could keep the service bits unchanged and only update the in browser paths to use the jrassign. Would you prefer that?

json-web-token should never really have been used in a browser context, webpack just happened to bial us out.

Reply:
testing should be free, as long as we have mocha test that exercise it, as mocha runs in node :)

Reply:
> We could keep the service bits unchanged and only update the in browser paths to use the jrassign. Would you prefer that?

Yeah I would prefer that just to be safe.

> json-web-token should never really have been used in a browser context, webpack just happened to bial us out.

Totally agree.

Reply:
@tanviraumi/ @anthony-murphy  - it looks like tree shaking doesn't work for the json-web-token + server-services-client combo, so we either need to modify server-services-client or break the dependency server-local-server has on server-services-client by perhaps moving GitManager and the name generation modules out into another package. None of these options are great IMO. Thoughts on the best approach here to unblock webpack 5 for consumers of FF?

Reply:
Another alternative is to have two different versions of signing token. We move the new jsrassign one to a client specific package and move the jsonwebtoken one to a service specific one (services/services-shared/services-utils). So they will both use their own version. Thoughts?

Reply:
That sounds reasonable, thanks for the suggestion. I'll take a look later today at the server side PR